chromium_src/net/cookies/cookie_monster.cc

```diff
@@ -104,7 +104,8 @@ void CookieMonster::SetCanonicalCookieAsync(
     std::unique_ptr<CanonicalCookie> cookie,
     const GURL& source_url,
     const CookieOptions& options,
-    SetCookiesCallback callback) {
+    SetCookiesCallback callback,
+    const CookieAccessResult* cookie_access_result) {
```

Comment:
shouldn't we be sending this param to the ephemeral monster `SetCanonicalCookieAsync` method?

Reply:
Good catch... the `ephemeral_monster` is a `ChromiumCookieMonster` as well, so I think you're right in that we need to pass it there, too, otherwise it will be using the default `nullptr`. I've made the change locally and run all the unit tests plus all the EphemeralStorage-related browser tests and everything passes, so I pushed the fix to cr100 in https://github.com/brave/brave-core/commit/2cfb3bc01c43846fd76a1f793bd559988584d209